doc/src/explanation/glossary.md

```diff
@@ -0,0 +1,712 @@
+    instance typically cannot be used in SymPy functions, unless it can be
+    turned into one via {term}`sympify()`.
+
+{class}`~.Boolean`
+
+    *{class}`~.Boolean`* is the base class for the classes in the
+    {mod}`~.logic` module. `Boolean` instances represent logical predicates
+    that are elements of a [boolean
+    algebra](https://en.wikipedia.org/wiki/Boolean_algebra) and can be thought
+    of as having a "true" or "false" value (note that `Boolean` objects do not
+    use the {term}`three-valued logic` used by the {term}`assumptions`).
+
+Canonical Form
+Canonicalize
+
+    Often expressions can be written in multiple, mathematically equivalent
+    ways. A *canonical form* is a single way of writing an expression, which
+    all equivalent expressions can be transformed to. An expression that is
+    put into a canonical form is said to be *canonicalized*. Often canonical
+    forms are unique and have properties that make them easier to work with.
+    For example, a common canonical form used for rational functions is
+    $\frac{p}{q}$, where $p$ and $q$ are expanded polynomials with no common
+    factors.
+
+Code Generation
+
+    *Code generation* refers to the process of taking a SymPy expression and
+    converting it into code for a language or library so that it can be
+    evaluated numerically. SymPy supports code generation for [dozens of
+    languages](codegen_module) and libraries including C, C++, Fortran, and
+    NumPy.
+
+Core
+
+    The [*core*](core_module) is the submodule that contains the important
+    functionality used by all SymPy objects. This includes the {term}`Basic`
+    and {term}`Expr` base classes, classes like {class}`~.Add`,
+    {class}`~.Mul`, and {class}`~.Pow`, and the {term}`assumptions`.
+
+Equation
+
+    An *equation* is an {term}`expression` that has an equals sign $=$.
+    Equations in SymPy are represented using the {class}`Eq
+    <sympy.core.relational.Equality>` class. Equations are **not** created
+    using the `==` operator. The `==` operator does a {term}`structural
+    equality` check between two expressions, and always returns `True` or
+    `False`. To contrast, a symbolic equation may be {term}`unevaluated`.
+    Equations are considered {term}`booleans <boolean>` since they
+    mathematically represent a predicate value that is either true or false.
+
+`_eval_*`
+
+    Various methods on {term}`Basic` and {term}`Expr` can be defined on
+    subclasses using special *`_eval_*`* methods. For example, an object can
+    define how it will be processed by the {func}`~.diff` function by defining
+    a `_eval_derivative` method. `_eval_*` methods used are instead of
+    overriding the method itself so that the method defined on the base class
+    can do pre-processing before dispatching to the `_eval_*` method.
+
+`evalf`
+
+    [*`evalf`*](sympy.core.evalf.EvalfMixin.evalf) is the method present on
```

Comment:
I think it's worth mentioning `.n()` and `N()` here. It's not obvious that these are all equivalent.